components/dashboard/wordcloud.tsx

```diff
@@ -1,15 +1,50 @@
 import { Box, Grid } from "@mui/material"
 import React, { useContext, useMemo, useState } from "react"
 import { DashboardFilterContext } from "./context"
-import ReactWordcloud, { Word } from "react-wordcloud"
+import ReactWordcloud, { CallbacksProp, OptionsProp, Word } from "react-wordcloud"
 import { WordcloudPopover } from "./wordcloudPopover"
 // import { sticky } from "tippy.js"
 
-export const Wordcloud: React.FC<{ words?: Word[] }> = ({ words }) => {
+export const Wordcloud: React.FC<{ words?: Word[] }> = ({ words: wordsProp }) => {
   const [popoverOpen, setPopoverOpen] = useState(false)
   const [popoverEntity, setPopoverEntity] = useState("")
   const { sentenceTerms } = useContext(DashboardFilterContext)
 
+  const options: OptionsProp = useMemo(() => {
+    return {
+      deterministic: true,
+      fontSizes: [24, 72],
+      svgAttributes: { style: "transform: translate(-120px, -100px);" },
+      fontFamily: "arial-black",
+      // these tippy props have no effect. reaching out to the authors of react-wordcloud about this issue
+      // tooltipOptions: {
+      //   interactive: true,
+      //   interactiveBorder: 100,
+      //   interactiveDebounce: 100,
+      //   appendTo: () => document.body,
+      //   sticky: true,
+      //   plugins: [sticky],
+      // },
+    }
+  }, [])
+
+  const callbacks: CallbacksProp = useMemo(() => {
+    return {
+      getWordColor: (word) =>
+        sentenceTerms.includes(word.text)
+          ? "lime"
+          : ["red", "blue", "brown", "purple"][word.value % 4],
+      onWordClick: (word) => {
+        setPopoverOpen(true)
+        setPopoverEntity(word.text)
+      },
+    }
+  }, [])
+
+  const words: Word[] | undefined = useMemo(() => {
+    return wordsProp
+  }, [wordsProp])
+
```

Comment:
I memoized all these props to prevent the component from rerendering so often. this was happening due to the props being arrays and objects which are more sensitive to triggering rerender